src/Cli/dotnet/Installer/Windows/MsiPackageCache.cs

```diff
@@ -143,12 +165,14 @@ public void CachePayload(string packageId, string packageVersion, string manifes
 
             if (IsElevated)
             {
-                CreateRootDirectory();
+                // Create the root directory. If the directory already exists then we'll try to
+                // reset the DACL to conform to our expectations.
+                CreateSecureDirectory(PackageCacheRoot, resetAccessRules: true, Log);
 
                 string packageDirectory = GetPackageDirectory(packageId, packageVersion);
 
-                // Delete the directory and create a new one that's secure. If the files were properly
-                // cached, the client won't request this action.
+                // Delete the package directory and create a new one. If all the files were properly
```

Comment:
Did you remove "that's secure" because that isn't important for the MSI payload?